### PR TITLE
RAC-480: Add TextAreaInput in DSM

### PR DESCRIPTION
**Pull request by ValentinMumble**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by juliensnz — APPROVED** [review]

**Review by juliensnz — APPROVED** [review]

File: akeneo-design-system/package.json [hunk 1]
```diff
@@ -37,17 +39,21 @@
...
     "@types/jest-image-snapshot": "^3.1.0",
     "@types/puppeteer": "^3.0.1",
     "@types/react": "^17.0.0",
     "@types/react-dom": "^17.0.0",
+    "@types/react-draft-wysiwyg": "^1.13.1",
     "@types/styled-components": "^5.1.2",
     "@types/testing-library__jest-dom": "^5.9.2",
     "@typescript-eslint/eslint-plugin": "^4.1.0",
     "@typescript-eslint/parser": "^4.1.0",
+    "draft-js": "^0.11.7",
```
Comment (ValentinMumble) [inline akeneo-design-system/package.json:56]: In devDependencies or regular dependencies?

File: akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      'The giant panda, also known as the panda bear or simply the panda, is a <b>bear native</b> to south-central China. It is characterized by large, black patches around its eyes, over the ears, and across its round body. The name "giant panda" is sometimes used to distinguish it from the red panda, a neighboring musteloid. Though it belongs to the order Carnivora, the giant panda is a folivore, with bamboo shoots and leaves making up more than 99% of its diet. Giant pandas in the wild will occasionally eat other grasses, wild tubers, or even meat in the form of birds, rodents.',
+    placeholder: 'Please enter a value in the TextAreaInput',
+  }}
+/>
+
+# Text input (Beta)
+
+## Usage
+
+Text input allows the user to enter content and data when the expected user input is a single line of text.
```
Comment (ValentinMumble) [inline akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.stories.mdx:31]: Wrong description text here

File: akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    },
+  }}
+  args={{
+    value:
+      'The giant panda, also known as the panda bear or simply the panda, is a <b>bear native</b> to south-central China. It is characterized by large, black patches around its eyes, over the ears, and across its round body. The name "giant panda" is sometimes used to distinguish it from the red panda, a neighboring musteloid. Though it belongs to the order Carnivora, the giant panda is a folivore, with bamboo shoots and leaves making up more than 99% of its diet. Giant pandas in the wild will occasionally eat other grasses, wild tubers, or even meat in the form of birds, rodents.',
+    placeholder: 'Please enter a value in the TextAreaInput',
+  }}
+/>
+
+# Text input (Beta)
```
Comment (ValentinMumble) [inline akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.stories.mdx:27]:
```suggestion
# TextArea input (Beta)
```

File: akeneo-design-system/src/components/Field/Field.tsx [hunk 4]
```diff
@@ -93,7 +101,7 @@ const Field = React.forwardRef<HTMLDivElement, FieldProps>(
     });
 
     return (
-      <FieldContainer ref={forwardedRef} {...rest}>
+      <FieldContainer ref={forwardedRef} fullWidth={fullWidth ?? false} {...rest}>
```
Comment (ValentinMumble) [inline akeneo-design-system/src/components/Field/Field.tsx:104]:
```suggestion
      <FieldContainer ref={forwardedRef} fullWidth={fullWidth} {...rest}>
```

File: akeneo-design-system/src/components/Field/Field.tsx [hunk 5]
```diff
@@ -66,6 +66,11 @@ type FieldProps = {
    */
   channel?: string | null;
 
+  /**
+   * Should the field input take the full width of the parent container
```
Comment (ValentinMumble) [inline akeneo-design-system/src/components/Field/Field.tsx:70]:
```suggestion
   * Should the field input take the full width of the parent container.
```

File: akeneo-design-system/src/components/Input/TextAreaInput/RichTextEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  return EditorState.createWithContent(ContentState.createFromBlockArray(rawDraft.contentBlocks));
+};
+
+type RichTextEditorProps = {
+  value: string;
+  readOnly?: boolean;
+  onChange: (value: string) => void;
+};
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Input/TextAreaInput/RichTextEditor.tsx:24]: Overide EditorProps to allow user send other options ? Like customBlockRenderFunc, we used it in AssetManager but i don't know why

File: akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.tsx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+  & .rdw-editor-main {
+    min-height: 200px;
+    max-height: 400px;
+  }
+
+  & .rdw-editor-toolbar {
+    border: none;
+    padding: 0;
+    margin: 0;
+    /* margin-left: -9px; */
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.tsx:47]: Can be removed ?

File: akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.unit.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    <>
+      <label htmlFor="myInput">My label</label>
+      <TextAreaInput id="myInput" value="Nice RTF content" isRichText={true} onChange={handleChange} />
+    </>
+  );
+
+  expect(screen.getByLabelText('rdw-wrapper')).toBeInTheDocument();
+  expect(screen.getByText('Nice RTF content')).toBeInTheDocument();
+});
+
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.unit.tsx:81]: Maybe add on change and readonly validation in rich text editor

File: akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.tsx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+    box-shadow: 0 0 0 2px ${getColor('blue', 40)};
+  }
+`;
+
+const RichTextEditorContainer = styled.div<{readOnly: boolean; invalid: boolean} & AkeneoThemedProps>`
+  ${CommonStyle}
+  padding: 0;
+  padding-bottom: 10px;
+
+  & .rdw-editor-main {
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.tsx:38]: :sob: :smile:

**Review by StevenVAIDIE — APPROVED** [review]

File: akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.tsx (hunk 9 above)
Comment (ValentinMumble) [reply akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.tsx:38]: :sad-dog:

File: akeneo-design-system/src/components/Input/TextAreaInput/RichTextEditor.tsx (hunk 6 above)
Comment (ValentinMumble) [reply akeneo-design-system/src/components/Input/TextAreaInput/RichTextEditor.tsx:24]: I agree, we may need to transmit the props to the editor 👍

File: akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.unit.tsx (hunk 8 above)
Comment (juliensnz) [reply akeneo-design-system/src/components/Input/TextAreaInput/TextAreaInput.unit.tsx:81]: I tried but didn't succeded to implement this test. We will see together how to add it

File: akeneo-design-system/src/components/Input/TextAreaInput/RichTextEditor.tsx (hunk 6 above)
Comment (juliensnz) [reply akeneo-design-system/src/components/Input/TextAreaInput/RichTextEditor.tsx:24]: I tried but there is type incompatibilities between react-draft-js and draft-js. We will see together next week

**Pull request merged**